Create propagation_p368.py

fgiovanard opened the pull request.

fgiovanard committed:
Create propagation_p368.py

bwinkel commented:
Dear @fgiovanard, 

thanks a lot for your first Pull Request! There a a few minor things, which we'll need to fix before merging; see comments.

A general remark, please make a few changes to the formatting. I like to stick to pep8 as much as possible. If you want a quick fix, simply use [black](https://github.com/psf/black) to reformat it automatically. I think, you could also simply use this [online tool](https://black.vercel.app/) for reformatting.

Let me know if you need help.

fgiovanard committed:
continue working p368

fgiovanard committed:
Update on low freq propagation with tests

Integrate ITU-R SM.2028-0 / P.368 low-frequency propagation module into pycraf's pathprof package

Refactor findE40, low_prop, and protection_distance functions for computing protection distance between inductive systems and radio services below 30 MHz
Vectorize field/distance maths using ranged_quantity_input to handle arrays in a single call
Fix N×N broadcasting bug in regime selection using np.choose, restoring correct (N,) output shape
Correct Table 2 permittivity for average-salinity seawater (70 → 80) per ITU Handbook on Ground Wave Propagation (2014)
Add test suite with 10 cases, including vectorized-shape regression guard
